compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java

```diff
@@ -131,7 +131,6 @@ public BIRTypeWriter(ByteBuf buff, ConstantPool cp, Env typeEnv) {
     }
 
     public void visitType(BType type) {
-        writeSemType(type.semType());
```

Comment:
Now we don't serialize the semtype? We serialize/deserialize BType and we re-construct the semtype from that deserialized BType right?

Reply:
Mostly yes ([updated](https://github.com/ballerina-platform/ballerina-lang/issues/42924#issuecomment-2229867817) the issue with the rational behind this)